loadtest/user/userentity/websocket.go

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) 2019-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package userentity
+
+import (
+	"time"
+
+	"github.com/mattermost/mattermost-server/v5/mlog"
+	"github.com/mattermost/mattermost-server/v5/model"
+)
+
+const (
+	// Same as the webapp settings.
+	minWebsocketReconnectDuration = 3 * time.Second
+	maxWebsocketReconnectDuration = 5 * time.Minute
+	maxWebsocketFails             = 7
+)
+
+// listen starts to listen for messages on various channels.
+// It will keep reconnecting if the connection closes.
+// Only on calling Disconnect explicitly, it will return.
+func (ue *UserEntity) listen() {
+	connectionFailCount := 0
+start:
+	client, err := model.NewWebSocketClient4(ue.config.WebSocketURL, ue.client.AuthToken)
```

Comment:
Do we need to create a new client every time we disconnect? Calling the `Listen()` method should be enough.

Reply:
Yes, I believe so. When the eventchannel and responsechannel are closed, the underlying connection is also closed. And `Listen` just starts to do `if _, rawMsg, err = wsc.Conn.ReadMessage(); err != nil {`. Which won't work if the connection is closed. We need to create a new one.

I have tested the code by shutting down the server, trying to reconnect and verified that it succesfully reconnects when the server comes up again.

Reply:
Mmm, interesting. Can't we just call `WebSocketEvent.Connect()` if we disconnect?

Reply:
Ah yes, that works too :+1: Sorry, hadn't seen that method. :stuck_out_tongue: